coop_shift/model/res_partner.py

```diff
@@ -382,20 +390,20 @@ def _compute_is_exempted(self):
     @api.depends(
         'extension_ids.date_start', 'extension_ids.date_stop',
         'extension_ids.partner_id')
-    def compute_date_delay_stop(self):
-        """This function should be called in a daily CRON"""
+    def _compute_date_delay_stop(self):
+        """Function should be called in a daily CRON."""
         for partner in self:
             max_date = False
             for extension in partner.sudo().extension_ids:
-                if extension.date_start <= fields.Datetime.now() and\
-                        extension.date_stop > fields.Datetime.now():
-                    max_date = max(max_date, extension.date_stop)
+                if extension.date_start <= fields.Date.today(
+                ) and extension.date_stop > fields.Date.today():
```

Comment:
When you have long if statements, it's usually best to try something like this (it's more readable):

```py
if (
    extension.date_start <= fields.Date.today()
    and extension.date_stop > fields.Date.today()
):
    # do something
    # ...
```